src/SonarScanner.MSBuild.PreProcessor/ProcessedArgs.cs

```diff
@@ -247,7 +246,7 @@ private ServerInfo GetAndCheckServerInfo(ILogger logger, bool isHostSet, string
                 { isHostSet: true, isSonarcloudSet: true } => Warn(new(sonarcloudUrl, defaultSonarCloudApi, true), Resources.WARN_HostUrlAndSonarcloudUrlSet),
                 { isHostSet: false, isSonarcloudSet: false } => new(defaultSonarCloud, defaultSonarCloudApi, true),
                 { isHostSet: false, isSonarcloudSet: true } => new(sonarcloudUrl, defaultSonarCloudApi, true),
-                { isHostSet: true, isSonarcloudSet: false } => sonarHostUrl == defaultSonarCloud
+                { isHostSet: true, isSonarcloudSet: false } => sonarHostUrl.TrimEnd('/') == defaultSonarCloud
```

Comment:
Shouldn't we do this for all the urls?

Reply:
The pseudocode I linked doesn't seem to care.
From our POV, we set this to the `HttpClient.BaseAddress` of the `IDownloader`, so we don't care either way.
Here I do it to make sure that the two URIs match.

Reply:
I'm not sure if we're affected, but we need to be careful with the base URLs. The HttpClient uses the Uri class for concatenation, and due to how it works, this can be surprising. 

See the remarks from: https://learn.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-8.0#system-uri-ctor(system-uri-system-string)

We might want to add tests for the get version scenario `return await GetVersion(downloader|fallback, "api/server/version")` when the base also contains a subpath.

Reply:
We checked, and `WebClientDownloader` always adds a slash at the end of the base URL.

https://github.com/SonarSource/sonar-scanner-msbuild/blob/master/src/SonarScanner.MSBuild.Common/WebUtils.cs#L32